src/Github/CreateReleaseTextViaKeepAChangelog.php

```diff
@@ -84,4 +128,78 @@ private function fetchChangelogContentsFromBranch(
 
         return $contents;
     }
+
+    /**
+     * @psalm-param non-empty-string $changelog
+     * @psalm-param non-empty-string $version
+     * @psalm-return non-empty-string
+     */
+    private function updateReleaseDate(string $changelog, string $version): string
+    {
+        $lines = explode("\n", $changelog);
+        Assert::greaterThan(count($lines), 0);
+
+        $releaseLine = $lines[0];
+        $regex       = sprintf('/^(## (?:%1$s|\[%1$s\])).*$/i', preg_quote($version));
+        $lines[0]    = preg_replace($regex, '$1 - ' . $this->clock->now()->format('Y-m-d'), $releaseLine);
+
+        return implode("\n", $lines);
+    }
+
+    /**
+     * @psalm-param non-empty-string $changelog
+     * @psalm-return non-empty-string
+     */
+    private function removeDefaultContents(string $changelog): string
+    {
+        $contents = str_replace(self::DEFAULT_CONTENTS, '', $changelog);
+        Assert::notEmpty($contents);
+
+        return $contents;
+    }
+
+    /**
+     * @psalm-param non-empty-string $contents
+     * @psalm-param non-empty-string $version
+     */
+    private function fetchChangelogEntry(string $contents, string $version): ChangelogEntry
+    {
+        $entryContents = [];
+        $entryIndex    = null;
+        $entryLength   = 0;
+        $boundaryRegex = '/^(?:## (?:\d+\.\d+\.\d+|\[\d+\.\d+\.\d+\])|\[.*?\]:\s*\S+)/i';
+        $regex         = sprintf('/^## (?:%1$s|\[%1$s\])/i', preg_quote($version));
+
+        foreach (explode("\n", $contents) as $index => $line) {
```

Comment:
Similar to other changelog normalisation logic, this is really hard to follow.

Can defer refactoring, but need comments about what is being done in this method.